index.js

```diff
@@ -10,9 +10,93 @@ function ask(questionText) {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
+
+  // Define constant range to start game:
+  let min = 1;
+  let max = 101;
+
+  //Greet user and offer game rules:
+  console.log("Hello. I am Lieutenant Commander Data.\nPlease think of a number between 1 and 100 (inclusive).\nI will try to guess it.")
+
+  //Gather user response if they have thought of a number within the range; display response for information:
+  let randNumResp = await ask("Have you selected a random number between 1 and 100? (Y/N)\n");
+
+  console.log('You entered: ' + randNumResp);
+
+  if (randNumResp === 'N') {
+    console.log("I'll assume you've chosen by now.");
+
+  };
+
+  //Declare and initialize app-level variables:
+  let highLow;
+  let middle;
+  let lastGuess;
+  let nextGuess = Math.floor((max + min) / 2);
+  let tryCount = 1;
+  // Ask user if guess is correct; first run will be for range set in app-level variables:
+  let userChoice = (await ask("Is it... " + nextGuess + "? (Y/N)\n")).toUpperCase();
+
+  ////////////////////////////////////////////
+
+  // Declare function guessNum() to calculate binary search:
+  /*
+    pivot is equal to the number known to not be the correct answer within the range.
+    option is the user input as to whether the game should guess higher or lower.
+    middle is set to be the midpoint in the range defined by the pivot and either the max or min as defined prior
+  */
+  function guessNum(pivot, option) {
+
+    if (option === 'H') {
+      middle = Math.floor((max + pivot) / 2);
+    } else if (option === 'L') {
+      middle = Math.floor((pivot + min) / 2);
+    };
+    return middle;
+  }
+
+  ////////////////////////////////////////////////
+
+
+  // Start while loop to generate successive guesses:
+  while (userChoice !== 'Y') {
+
+
+    if (userChoice === 'N') {
+
+      highLow = (await ask('Hmmm...is it higher(H) or lower(L)?')).toUpperCase();
+
+      if (highLow === 'H') {  // evaluate the user's highLow input for meaning 'go higher' or 'go lower'
+        min = nextGuess;      // set the value of min to that of the prior guess
+      } else if (highLow === 'L') {
+        max = nextGuess;     // set the value of max to that of the prior guess
+      };
+
+      // Call the guessNum() function with the prior guess in the pivot param, highLow input in the option param:
+      nextGuess = guessNum(nextGuess, highLow);   // set the function result to be the next guess
+      
+// Evaluate whether user has forgotten or changed the original number:
+      if (nextGuess === lastGuess && userChoice === 'N') {
+        console.log("But you said your number was above " + lastGuess + "!\nHow can it now be below " + nextGuess + "?\nData out.");
+        process.exit();
+
+      };
```

Comment:
There is some funky behavior happening here where you are checking if the user is cheating. This works for the case where a user first says that their number is above a certain guess, but then tries to go lower, but not the opposite. Try to enter some input into your program such that it catches the user "cheating" and check out what's happening. YOu could try a check that is something like this:
```js
if (nextGuess <= min) {
  // guess can't be lower than the already determined minimum
}
if (nextGuess >= max {
  //guess can't be higher than the already determined maximum
}
```

Reply:
When I run the program, I am asked if the number is 51 or not, and whether it is higher or lower than 51. Either way, if I choose higher or lower at this prompt, I am then asked questions which successively narrow down the range. When I get to the last step, which is step 7, if I contradict the only number it could be, the program tells me there is no way it can be that, because this contradicts the first high/low answer. I'm not sure where you are seeing it fail. It does not catch it quickly, but it does seem to catch it. Could you clarify how it is supposed to run?